src/frontend/src/env/types/env-icrc-token.ts

```diff
@@ -29,3 +30,12 @@ export type EnvIcrcTokenMetadata = z.infer<typeof envIcrcTokenMetadata>;
 export type EnvIcrcToken = z.infer<typeof envIcrcToken>;
 
 export type EnvIcrcTokens = z.infer<typeof envIcrcTokens>;
+
+const envAdditionalIcrcTokens = z.record(envTokenSymbol, z.union([z.undefined(), envIcToken]));
```

Comment:
I'm a bit out of context, probably why I don't get it but, why undefined here?

Reply:
Aiming for this structure: https://github.com/dfinity/oisy-wallet/blob/de85087df27d0ce3267cacfad8740046da7065d2/src/frontend/src/env/types/env-token-ckerc20.ts#L11C1-L12C22

Reply:
Gotcha! In another PR, we should extract a generator/schema for the union `z.union([z.undefined(), T])`.
This way we avoid duplicate code and we can express the goal of the `undefined`.

Reply:
Ok, I add a todo for this and will address it. Although TBH I'm unsure what the goal of the undefined is. :D